bigtable/bigtable-proxy/src/main/java/com/google/cloud/bigtable/examples/proxy/channelpool/DataChannel.java

```diff
@@ -182,7 +182,9 @@ private ListenableFuture<PingAndWarmResponse> sendPingAndWarm(PingAndWarmRequest
         new Listener<>() {
           @Override
           public void onMessage(PingAndWarmResponse response) {
-            f.set(response);
+            if (!f.set(response)) {
+              LOGGER.warn("PingAndWarm returned multiple responses");
```

Comment:
Hmm, should we error out if this happens? or record a metric?

Reply:
We cant error out because the future is already set. Metric would be a good idea. I added a TODO...this PR is already getting too big :(